src/sdks/core/src/cpp/sdk/cpptest/unit/accessibilityTest.cpp

```diff
@@ -0,0 +1,314 @@
+{
+protected:
+    JsonEngine *jsonEngine;
+    Firebolt::Error error = Firebolt::Error::None;
+
+    void SetUp() override
+    {
+        jsonEngine = new JsonEngine();
+    }
+
+    void TearDown() override
+    {
+        delete jsonEngine;
+    }
+
+    std::string fontFamilyToString(Firebolt::Accessibility::FontFamily fontFamily)
+    {
+        std::string str = "";
+        switch (fontFamily)
+        {
+        case Firebolt::Accessibility::FontFamily::MONOSPACED_SERIF:
+            str = "monospaced_serif";
+            break;
+        case Firebolt::Accessibility::FontFamily::PROPORTIONAL_SERIF:
+            str = "proportional_serif";
+            break;
+        case Firebolt::Accessibility::FontFamily::MONOSPACED_SANSERIF:
+            str = "monospaced_sanserif";
+            break;
+        case Firebolt::Accessibility::FontFamily::PROPORTIONAL_SANSERIF:
+            str = "proportional_sanserif";
+            break;
+        case Firebolt::Accessibility::FontFamily::SMALLCAPS:
+            str = "smallcaps";
+            break;
+        case Firebolt::Accessibility::FontFamily::CURSIVE:
+            str = "cursive";
+            break;
+        case Firebolt::Accessibility::FontFamily::CASUAL:
+            str = "casual";
+            break;
+        default:
+            str = "unknown";
+        }
+        return str;
+    }
+
+    std::string fontEdgeToString(Firebolt::Accessibility::FontEdge fontEdge)
+    {
+        std::string str = "";
+        switch (fontEdge)
+        {
+        case Firebolt::Accessibility::FontEdge::NONE:
+            str = "none";
+            break;
+        case Firebolt::Accessibility::FontEdge::RAISED:
+            str = "raised";
+            break;
+        case Firebolt::Accessibility::FontEdge::DEPRESSED:
+            str = "depressed";
+            break;
+        case Firebolt::Accessibility::FontEdge::UNIFORM:
+            str = "uniform";
+            break;
+        case Firebolt::Accessibility::FontEdge::DROP_SHADOW_LEFT:
+            str = "drop_shadow_left";
+            break;
+        case Firebolt::Accessibility::FontEdge::DROP_SHADOW_RIGHT:
+            str = "drop_shadow_right";
+            break;
+        default:
+            str = "unknown";
+        }
+        return str;
+    }
+};
+
+TEST_F(AccessibilityTest, ClosedCaptions)
+{
+    nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Accessibility.closedCaptions"));
+
+    Firebolt::Accessibility::ClosedCaptionsSettings closedCaptions = Firebolt::IFireboltAccessor::Instance().AccessibilityInterface().closedCaptions(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve closedCaptions from Accessibility.closedCaptions() method";
+    EXPECT_EQ(closedCaptions.enabled, expectedValues["enabled"]);
+
+    if (closedCaptions.styles.has_value()) {
+        const auto& styles = closedCaptions.styles.value();
+
+        if (styles.backgroundColor.has_value())
```

Comment:
What if styles.background doesn't have a value? Will this pass or fail? Is there a way to enforce that if our example has "styles.backgroundColor" then the result should and fail if that's not the case? This applies to all other elements as well.

Reply:
You're correct that the current implementation does not explicitly enforce that styles.backgroundColor and other fields must exist in both the actual result and the expected values. If the actual result is missing a field while the expected values contain it, the test will simply skip the EXPECT_EQ comparison, which could potentially lead to false positives if the field was expected but not present.

We could maybe do something like this:
```
if (expectedValues["styles"].contains("backgroundColor")) {
    ASSERT_TRUE(styles.backgroundColor.has_value()) << "Expected backgroundColor, but it is missing";
    EXPECT_EQ(styles.backgroundColor.value(), expectedValues["styles"]["backgroundColor"]);
}
```